gtfs-realtime/spec/en/trip-modifications.md

```diff
@@ -0,0 +1,65 @@
+# Trip Modifications
+
+A `TripModifications` message identifies a list of similar `trip_ids` from the (CSV) GTFS sharing the same stop pattern, which are all affected by particular modifications, such as a detour.
+
+## SLO: Service-level objective
+
+The frequency of data updates is expected to be approximately hourly (~24 times/day). Ingestion time may depend on the total number of affected trips. Consumers are expected to ingest a single TripModification within 5 minutes, and a feed with hundreds of detours within 20 minutes.
+
+## TripModifications
+
+The `TripModifications` is in effect on all of the listed service\_dates, until it is removed from the feed. On any given service date, a trip MUST NOT be assigned to more than one `TripModifications` object.
+
+There MAY be multiple `TripModifications` for a given stop pattern. It may be desirable to split the trips into multiple modifications e.g. if the `propagated_modification_delay` changes significantly, over the course of the detour.
+
+The trips created through GTFS-TripModifications modify and replace each specified `trip_id`, and don't create a copy or additional run. Modifications are applied on the schedule information, like if a static GTFS (CSV) was modified. 
+
+The scheduled stop times of each replacement trip are created from those of the affected trip, by performing the changes listed in modifications. `stop_sequence` for all stop times are replaced by a new value of 1 to n, starting with 1 on the first stop_time and increasing by 1 for each stop in the trip. A `TripUpdate` message must be provided to publish real-time arrival/departure times for the replacement trip.
+
+
+## Linkage to TripUpdates
+
+* A TripUpdate SHOULD be provided using a `ModifiedTripSelector` inside the TripUpdate's `TripDescriptor`. 
+    * When the TripUpdate refers to the replacement trip, the consumer should behave as if the static GTFS would have been modified with the TripModifications (e.g. `arrival_time`, `departure_time`, `stop_sequence`, `stop_id` on replacement stops)
+    * When providing a `ModifiedTripSelector`, the other fields of the `ScheduleRelationship` SHOULD be left empty, to avoid confusion by consumer that aren't looking for the `ModifiedTripSelector` value. 
```

Comment:
Would it be an error to specify `trip_id` and `modified_trip` in the same TripDescriptor?

Reply:
I felt it was maybe a bit strong to disallow it entirely because it's not semantically wrong, but maybe it's better to enforce backward compatibility? It would be disallowed for `trip_id` but discouraged for other fields I think.  I don't have a strong opinion

Reply:
I would disallow it for better backward compatibility unless there is a use case I'm not thinking of where providing both `trip_id` and `modified_trip` would be needed.

@gcamp Should "the other fields of the `ScheduleRelationship` SHOULD be left empty" be changed to "the other fields of the `TripDescriptor` SHOULD be left empty"? (And also change SHOULD to MUST if there is agreement on requiring this.)